scale-decode/src/impls/mod.rs

```diff
@@ -609,9 +637,12 @@ macro_rules! impl_decode_tuple {
         impl < $($t),* > DecodeAsFields for ($($t,)*)
         where $( $t: IntoVisitor, )*
         {
-            fn decode_as_fields<'info>(input: &mut &[u8], fields: &mut dyn FieldIter<'info>, types: &'info scale_info::PortableRegistry) -> Result<Self, Error> {
-                let mut composite = crate::visitor::types::Composite::new(input, crate::EMPTY_SCALE_INFO_PATH, fields, types, false);
-                let val = <($($t,)*)>::into_visitor().visit_composite(&mut composite, crate::visitor::TypeId(0));
+            fn decode_as_fields<'resolver, Resolver: TypeResolver>(input: &mut &[u8], fields: &mut dyn FieldIter<'resolver, Resolver::TypeId>, types: &'resolver Resolver) -> Result<Self, Error> {
+                let mut composite = crate::visitor::types::Composite::new(input, fields, types, false);
+
+                // [jsdw] TODO: Passing a "default type ID" to a visitor just to satisfy the signature is
```

Comment:
maybe file an issue for this?

Reply:
Thanks for the reminder; I planned to as soon as this is merged but it's easy to forget so I'll just do it now!

Reply:
https://github.com/paritytech/scale-decode/issues/46